framework/Optimizers/gradients/FiniteDifference.py

```diff
@@ -68,28 +74,50 @@ def evaluate(self, opt, grads, infos, objVar):
       @ Out, direction, dict, versor (unit vector) for gradient direction
     """
     gradient = {}
+    A=[]
+    #print("This is grads",grads[0].keys())
+    #print("This is opt",opt)
+    #aaa
+    #print("These are")
     for g, pt in enumerate(grads):
+      x, y= enumerate(grads)
+
       info = infos[g]
       delta = info['delta']
       activeVar = info['optVar']
-      lossDiff = np.atleast_1d(mathUtils.diffWithInfinites(pt[objVar], opt[objVar]))
-      #if delta != 0:
-      grad = (lossDiff) / delta
-      #else:
-      #  grad = 0
+
+
+      #CentralDifference.derivative(pt[objVar],opt[objVar],g,delta,grads,objVar,infos)
+
+
+      grad = self.derivative(pt[objVar],opt[objVar],delta,grads,objVar,infos)
```

Comment:
this doesn't work in the general case, it only works when all the variable values are 0-d floats. I recommend reverting changes to this file